scaper/core.py

```diff
@@ -358,6 +359,72 @@ def _validate_distribution(dist_tuple):
                 'number that is equal to or greater than trunc_min.')
 
 
+def _modify_source_time(source_time, source_duration, event_duration):
```

Comment:
How about `_ensure_valid_source_time_tuple`?

Reply:
not keen on including the term `valid` because scaper already has a notion of tuple validity (a tuple that satisfies the expected syntax), which is unrelated to whether the tuple can be satisfied given the specific constraints of a source file (I purposely avoided `valid` in my suggestions for this reason).

How about `_ensure_satisfiable_source_time_tuple`? It's a mouthful but it's clear.

Reply:
Yeah that's fine and makes sense. I'll edit!